app/services/two_factor_authentication/personal_key_configuration_manager.rb

```diff
@@ -0,0 +1,39 @@
+module TwoFactorAuthentication
+  class PersonalKeyConfigurationManager < TwoFactorAuthentication::ConfigurationManager
+    def enabled?
+      personal_key.present?
+    end
+
+    def configured?
+      personal_key.present?
+    end
+
+    def configurable?
+      false # we can always create a new personal key, but not as part of the 2fa process
+    end
+
+    ###
+    ### Method-specific data management
+    ###
+
+    delegate :active_profile, :personal_key, to: :user
+
+    # :reek:FeatureEnvy
+    def should_acknowledge?(session)
```

Comment:
This is another example of a method that better belongs in a Policy Object as opposed to a configuration manager. I think Reek is right in this case in pointing out that is belongs in a different class.

This used to be defined in `UserDecorator`, which we haven't been using responsibly. We've basically just been using it as an alternative dumping ground for methods that would otherwise have been placed in the User model. We should extract out the various concepts into individual Policy Objects.

Reply:
We could have a policy object for each 2fa method. Then the configuration manager manages the configuration and the policy object answers questions about that configuration, but not questions about aggregate 2fa methods. That would be for other policy objects.

I'll create a ticket so I can circle back around to the policy object pattern.